src/main/java/com/denizenscript/denizencore/objects/core/ImageTag.java

```diff
@@ -0,0 +1,386 @@
+        }
+        return valueOf(string, CoreUtilities.noDebugContext) != null;
+    }
+
+    public BufferedImage image;
+    public String imageType;
+
+    public ImageTag(BufferedImage image, String imageType) {
+        this.image = image;
+        this.imageType = imageType;
+    }
+
+    public static void register() {
+
+        // <--[tag]
+        // @attribute <ImageTag.pixel_at[x=<x>;y=<y>]>
+        // @returns ColorTag
+        // @description
+        // Returns the color of a specific pixel in an image.
+        // "x" and "y" are the position of the pixel, see <@link language Image positions>.
+        // @example
+        // Get the color of the pixel at 43,21
+        // - narrate "The color is: <[image].pixel_at[x=43;y=21]>."
+        // -->
+        tagProcessor.registerStaticTag(ColorTag.class, MapTag.class, "pixel_at", (attribute, object, param) -> {
+            ElementTag x = param.getRequiredObjectAs("x", ElementTag.class, attribute);
+            ElementTag y = param.getRequiredObjectAs("y", ElementTag.class, attribute);
+            if (x == null || y == null) {
+                return null;
+            }
+            if (!x.isInt() || !y.isInt()) {
+                attribute.echoError("Invalid x/y '" + x + '/' + y + "' specified: must be valid numbers.");
+                return null;
+            }
+            return ColorTag.fromARGB(object.image.getRGB(x.asInt(), y.asInt()));
+        });
+
+        // <--[tag]
+        // @attribute <ImageTag.width>
+        // @returns ElementTag(Number)
+        // @mechanism ImageTag.scale
+        // @description
+        // Returns the image's width (in pixels).
+        // @example
+        // Narrates an image's width.
+        // - narrate "The image is <[image].width> wide."
+        // -->
+        tagProcessor.registerStaticTag(ElementTag.class, "width", (attribute, object) -> {
+            return new ElementTag(object.image.getWidth());
+        });
+
+        // <--[tag]
+        // @attribute <ImageTag.height>
+        // @returns ElementTag(Number)
+        // @mechanism ImageTag.scale
+        // @description
+        // Returns the image's height (in pixels).
+        // @example
+        // Narrates an image's height.
+        // - narrate "The image is <[image].height> tall."
+        // -->
+        tagProcessor.registerStaticTag(ElementTag.class, "height", (attribute, object) -> {
+            return new ElementTag(object.image.getHeight());
+        });
+
+        // <--[tag]
+        // @attribute <ImageTag.type>
+        // @returns ElementTag
+        // @description
+        // Returns the image's type ("png", "jpg", "webp", etc.).
+        // @example
+        // Checks if an image is a png.
+        // - if <[image].type> == png:
+        //   - narrate "It's a png!"
+        // -->
+        tagProcessor.registerStaticTag(ElementTag.class, "type", (attribute, object) -> {
+            return new ElementTag(object.imageType, true);
```

Comment:
Why is this tracked on the image? This is a file property not an image property

Reply:
It's needed to write the image later (see `getRawBytes`, `image` command, etc.)

Reply:
Should be inputs in those locations then?